From aca6c34c8eae66c57794fbfd07c4c9eee520ce11 Mon Sep 17 00:00:00 2001
From: Nico Weber <thakis@chromium.org>
Date: Wed, 12 May 2021 18:21:25 +0000
Subject: [PATCH] build: Convert tools/json_to_struct to py3

Bug: 1205588
Change-Id: I2c61c32b8f1809a0109c63ad476ed2881ec30580
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2892470
Auto-Submit: Nico Weber <thakis@chromium.org>
Reviewed-by: Hans Wennborg <hans@chromium.org>
Commit-Queue: Nico Weber <thakis@chromium.org>
Cr-Commit-Position: refs/heads/master@{#882185}
---
 src/tools/json_to_struct/json_to_struct.gni | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/tools/json_to_struct/json_to_struct.gni b/src/tools/json_to_struct/json_to_struct.gni
index 8bf2e304e4..29afcb2106 100644
--- a/src/tools/json_to_struct/json_to_struct.gni
+++ b/src/tools/json_to_struct/json_to_struct.gni
@@ -2,8 +2,6 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import("//build/config/python.gni")
-
 # Converts a .json file to a C++ struct.
 #
 # Variables:
@@ -27,8 +25,7 @@ template("json_to_struct") {
   action_name = target_name + "_action"
   source_set_name = target_name
 
-  # TODO(crbug.com/1112471): Get this to run cleanly under Python 3.
-  python2_action(action_name) {
+  action(action_name) {
     visibility = [ ":$source_set_name" ]
     script = "//tools/json_to_struct/json_to_struct.py"
 
